msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,322 @@
+See comprehensive [messaging extension design guidelines](~/messaging-extensions/design/messaging-extension-design.md).
 
-  ![Help command sample](../../../../assets/images/faq/helpcommand.png)
+### 7.1 Action commands
 
-* **Use adaptive cards and task modules to make your bot response clear and actionable**
-[Adaptive cards with buttons invoking task modules](/task-modules-and-cards/task-modules/task-modules-bots) enhance the bot user experience. These cards and buttons are easier to use in a mobile device as opposed to your user typing the commands. Also bot responses must not be textual with long text. Bots must make use of adaptive cards and task modules instead of conversational chat based user interface and lengthy text responses.
+Action-based messaging extensions should do the following:
 
-* **Think through all scopes**. Be sure that your bot provides appropriate responses when mentioned (`@*botname*`) in a channel and in personal conversations. If your bot does not provide meaningful context within the personal or teams scope, disable that scope via the manifest. (See the `bots` block in the [Microsoft Teams manifest schema reference](../../../../resources/schema/manifest-schema.md#bots).)
+* Allow users to trigger actions on a message without specifying intermediate steps, such as signing in.
+* Pass the message context to the next work state.
 
-* **Include team, group chat, or 1:1 conversation**. Bot notifications must include a team, a group chat, or a one-to-one conversation with relevant content for your audience.
+### 7.2 Preview links (link unfurling)
 
-* **Do not push sensitive data**. Bots must not push sensitive data to a team, a group chat, or a 1:1 conversation, where there is an audience who must not view that data.
+Messaging extensions should preview recognized links in the Teams compose box. Do not add domains that are outside your control (either absolute URLs or wildcards). For example, `yourapp.onmicrosoft.com` is valid but `*.onmicrosoft.com` is not valid. Top-level domains also are prohibited (for example, `*.com` or `*.org`).
 
-* **Provide a welcome message**. Bot must provide an FRE welcome message that includes an interactive tutorial with carousel cards or "try it" buttons, to encourage engagement.
+See how to [implement link unfurling](~/messaging-extensions/how-to/link-unfurling.md).
 
-### &#9989; Personal bots must always send a welcome message on first launch
+### 7.3 Search commands
 
-A welcome message is the best way to set the tone for your personal chat bot. This is the first interaction a user has with the bot. A good welcome message can encourage the user to keep exploring the app. If the welcome or introductory message is confusing or unclear, users won't see the value of the app immediately and lose interest.
-See the following section for welcome message requirements:
+Search-based messaging extensions must provide text that helps users effectively search.
```

Comment:
Does this statement cover all types of search-based messaging extensions?

Reply:
Not clear. Let's discuss

Reply:
Yes it works for all search-based messaging extensions.